org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/util/PopupUtils.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * SonarLint for Eclipse
+ * Copyright (C) 2015-2023 SonarSource SA
+ * sonarlint@sonarsource.com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarlint.eclipse.ui.internal.util;
+
+import java.util.HashSet;
+
+import org.sonarlint.eclipse.ui.internal.notifications.AbstractNotificationPopup;
+
+/** Utility used for reducing stacked up notifications */
+public class PopupUtils {
```

Comment:
Was speaking to Marco about that. In the future we should handle it the same way for other pop-ups (the notifications) to lower the noise.